### DISCUSSION TITLE
Housekeeping old test items

### DISCUSSION THREAD

**Discussion by kierangirvan** [ludeknovy/jtl-reporter#109] [Ideas]
We run tests daily and have noticed some of the trend graphs only show the first 20 or so tests and not the most recent, the trend graph I'm referring to here is at transaction level i.e. wen you open up a test report, and view the request statistics and hit the trend icon towards the end of each row, it shows a nice trend of this transaction, but only shows the first 20 tests.

It would make more sense to me to either show all tests or the last 20 tests, and not the first.

That being said, in order to "workaround" this, we would like a mechanism to housekeep old tests, lets say for example a rule to delete all tests older than 20 days.  Is there such a mechanism, or can you provide a script or process to follow to safely remove older tests.

**Comment by ludeknovy** [comment]
Hi @kierangirvan , 
if it shows the first 20 tests, then its a bug - going to investigate that.

I like the idea of "housekeeping". Would it make sense to set it up at scenario or project level?

**Reply by kierangirvan** [reply]
I like the idea of housekeeping at scenario level.  As some scenarios may run at different frequencies for some users.

**Reply by ludeknovy** [reply]
The label order + sort fix is available in: novyl/jtl-reporter-be:v4.0.1

**Reply by kierangirvan** [reply]
I'll work with the team to get this upgraded in our live environment and confirm back once proven

**Reply by ludeknovy** [reply]
Cool! If upgrading from v3, please see https://github.com/ludeknovy/jtl-reporter#migration-from-v3-to-v4
And make sure you backup your v3 DB!

**Reply by ludeknovy** [reply]
https://user-images.githubusercontent.com/13610612/148644875-76e69451-fdc9-45fc-97f8-b9a6826d9351.mov

The retention setting is ready.

Now I only need to publish a new scheduler service to docker hub.

**Reply by ludeknovy** [reply]
The image is available here:  novyl/jtl-reporter-scheduler:v0.0.2

Needs to be tested.

**Reply by ludeknovy** [reply]
https://github.com/ludeknovy/jtl-reporter/releases/tag/v4.1.0
